cores/esp8266/WString.cpp

```diff
@@ -637,13 +614,14 @@ int String::lastIndexOf(char theChar) const {
 int String::lastIndexOf(char ch, unsigned int fromIndex) const {
     if (fromIndex >= len())
         return -1;
-    char tempchar = buffer()[fromIndex + 1];
-    wbuffer()[fromIndex + 1] = '\0';
-    char* temp = strrchr(wbuffer(), ch);
-    wbuffer()[fromIndex + 1] = tempchar;
+    char *wbuffer = this->wbuffer();
```

Comment:
The local variable `wbuffer` is the same as the class method `wbuffer()` so I'd really ask to change this name to something else.

I think the logic is still correct, it's just a readability and sanity thing.